.github/workflows/unitests.yml

```diff
@@ -36,6 +36,12 @@ jobs:
           version: ${{ matrix.vim }}
       - name: Checkout LSP plugin Code
         uses: actions/checkout@v4
+      - name: Prepare Tests
+        run: |
+          # install deps for the dummy lsp
+          cd ./test/dummy-lsp/
+          npm install
+          cd -
```

Comment:
perhaps just using markdown lsp? vs/but not need to make up a dummy lsp.

Reply:
I merely wanted a simple LSP setup for a very basic use case, keeping the unit tests equally simple.

The dummy LSP can be configured to return varying responses to hover requests, thus serving as a convenient way to mimic different LSPs using just one. Without this approach, you would have had to rely on two distinct LSP implementations—for example, starting with a PHP file, configuring both `phpactor` and `html-languageserver`, and checking whether the correct LSP responds upon querying an `htmlTag`. I find this complicates the tests unnecessarily.

Should implementing a fake LSP for simple mocking in tests prove problematic, replacing it with actual LSP instances wouldn't be an issue.

Do note that the dummy LSP could potentially be expanded for conducting additional tests. This idea serves as a suggestion rather than a requirement.

Reply:
you use a markdown file which embedded clang/golang etc code block to test, which those had been config into this test, if so, the code base would be clean and easy.

Reply:
I modified the unit tests according to recommendations, removing the dummy LSP. Although these tests aren't as extensive, they ensure the right LSP is invoked, which should be enough. Testing real LSP execution isn't feasible as clangd doesn't understand Markdown, and ultimately, testing real LSP execution turns out to be unnecessary because selecting the appropriate LSP suggests that everything else will most likely fall into place.